modules/services/compose/src/main/java/au/com/shiftyjelly/pocketcasts/compose/bottomsheet/ModalBottomSheet.kt

```diff
@@ -48,14 +55,23 @@ fun ModalBottomSheet(
         content = {}
     )
 
-    if (!sheetState.isVisible && shouldShow) {
-        displayBottomSheet(coroutineScope, sheetState)
-    }
     LaunchedEffect(Unit) {
         snapshotFlow { sheetState.currentValue }
             .collect {
                 if (sheetState.currentValue == ModalBottomSheetValue.Expanded) {
                     onExpanded.invoke()
+                } else if (sheetState.currentValue == ModalBottomSheetValue.Hidden) {
+                    if (isSheetShown) {
+                        /* Remove bottom sheet from parent view when bottom sheet is hidden
+                        on dismiss or back action for talkback to function properly. */
+                        parent.removeAllViews()
+                    } else {
+                        if (!sheetState.isVisible && shouldShow) {
+                            /* Show bottom sheet when it is hidden on initial set up */
+                            displayBottomSheet(coroutineScope, sheetState)
+                            isSheetShown = true
+                        }
+                    }
                 }
             }
     }
```

Comment:
`snapshotFlow` allowed me to observe bottom sheet visibility changes and react to show/hide actions [without having to cancel and re-launch `LaunchEffect`](https://developer.android.com/jetpack/compose/side-effects#launchedeffect). Your suggestion works as well, and tbh I don't have a clear answer at this point about which one has a clear advantage.

Reply:
That is interesting. Sounds like the gist of it is that reading a value in the key field of a `LaunchedEffect` does make that composable scope depend on that value, and it will recompose that entire scope anytime the key changes, even if the LaunchedEffect does not generate any UI. 

Thanks for pointing that out. I still have so much to learn with compose!